### ISSUE TITLE
Reuse idempotency key should be reused when restarting sync from a `429` or `5xx` response

### ISSUE THREAD

**Issue by nicdwilson** [woocommerce/woocommerce-square#106]
### Describe the bug

In troubleshooting 8051230-zd-a8c Square responded with the following regarding a sync interrupted by rate limiting.

> We are noticing that a new `idempotency_key` is being used in the retry attempts. Our best practices for idempotency recommend resubmitting the request using the same` idempotency_key` if you receive a `429 RATE_LIMITED` or `5XX HTTP` response code

---

## Product Requirements Section

> ***Dependency note:** This issue and* [SQUARE-272](https://linear.app/a8c/issue/SQUARE-272/rate-limit-retry-reuses-stale-idempotency-key-after-request-body) *are two sides of the same problem and must be solved together as a single idempotency key strategy. Neither should ship without the other.*

### **Problem & Expected Outcome**

**The Problem:** When a request fails with a `429` (rate limited) or `5xx` (server error) and the request body hasn't changed, Square's best practice is to retry with the same idempotency key. The plugin currently generates a new key every time, which violates this contract and can cause unexpected behavior on Square's side.

**The Fix:** As a developer, retries with an unchanged request body should reuse the original idempotency key, following Square's documented best practices.

### **Acceptance Criteria**

* On a `429` or `5xx` retry where the request body is unchanged, the original idempotency key is reused
* On a retry where the request body has changed, a new idempotency key is generated (see [SQUARE-272](https://linear.app/a8c/issue/SQUARE-272/rate-limit-retry-reuses-stale-idempotency-key-after-request-body))
* The decision of reuse vs. regenerate is made by comparing the request body, not the error type alone

### **Out of Scope**

* The inverse case (changed body on retry) — that is [SQUARE-272](https://linear.app/a8c/issue/SQUARE-272/rate-limit-retry-reuses-stale-idempotency-key-after-request-body) and must be solved in the same changeset
* Retry timing, backoff strategy, or maximum retry count changes

---

**Project classification:** Must Have

Review the [technical implementation document](<https://docs.google.com/document/d/1Uo6I77ch3Dh4-1bxY0X4_XELBczbCcswjheNI9pT-vY/edit?tab=t.0>) before starting work.

**Comment by nicdwilson** [comment]
p1713482527612829-slack-C7U3Y3VMY
